hostmap.go

```diff
@@ -537,6 +612,16 @@ func (hm *HostMap) addHostInfo(hostinfo *HostInfo, f *Interface) {
 			"hostinfo": m{"existing": true, "localIndexId": hostinfo.localIndexId, "hostId": hostinfo.vpnIp}}).
 			Debug("Hostmap vpnIp added")
 	}
+
+	i := 1
+	check := hostinfo
+	for check != nil {
+		if i > MaxHostInfosPerVpnIp {
+			hm.unlockedDeleteHostInfo(check)
+		}
+		check = check.next
```

Comment:
I think unlockedDeleteHostInfo will set check.next to nil, so do we need to re-order to grab check.next just in case its non-nil before unlockedDeleteHostInfo erases it?

In theory it can't be non-nil since the check would have run last time and prevented there being more entries I think.